File: gpp-api/src/main/java/com/develop/gpp/domain/entity/PecaModel.java
package com.develop.gpp.domain.entity;

import java.util.List;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import lombok.Getter;
import lombok.Setter;
@Setter
@Getter
@Entity
@Table(name = "peca")
public class PecaModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id_peca;

    @Column(nullable = false)
    private String numero;

    @Column(nullable = false)
    private String codigo_fabrica;

    @Column(nullable = false)
    private Integer unidade;

    @Column(nullable = false)
    private String descricao;

    @Column(nullable = false)
    private Integer altura;

    @Column(nullable = false)
    private Integer largura;

    @Column(nullable = false)
    private Integer profundidade;

    @Column(nullable = false)
    private Integer unidade_medida;

    @Column(nullable = false)
    private String volumes;

    @Column(nullable = false)
    private Boolean active;

    @Column(nullable = false)
    private Double custo;


    // @ManyToOne(fetch = FetchType.LAZY)
    // @JoinColumn(name = "id_peca_cor", referencedColumnName = "id_peca_cor", insertable = false, updatable = false)
    // private PecasCor pecasCor;



    @Column(nullable = false)
    private String cor;

    @Column(nullable = false)
    private String material;

    @Column(nullable = false)
    private Long id_fornecedor;

    // @OneToMany(mappedBy = "peca", cascade = CascadeType.ALL)
    // private List<PecasCor> pecasCorModel;

    @Column(nullable = false)
    private String material_fabricacao;

    // @OneToMany(mappedBy = "peca", cascade = CascadeType.ALL)
    // private List<PecasMaterial> pecasMaterialModel;

    // @OneToMany(mappedBy = "peca", cascade = CascadeType.ALL)
    // private List<PecasEspecieModel> pecasEspecieModel;

    /**
     *
     */
   @OneToMany(mappedBy = "peca")
    private List<ProdutoPecaModel> produtosPeca; 

}

File: gpp_app/lib/domain/models/PecaModel.dart
import 'ProdutoModel.dart';

class PecaModel {
  int? idPeca;
  String? numero;
  String? codigoFabrica;
  int? unidade;
  String? descricao;
  int? altura;
  int? largura;
  int? profundidade;
  int? unidadeMedida;
  String? volumes;
  bool? active;
  double? custo;
  String? cor;
  String? material;
  int? idFornecedor;
  String? materialFabricacao;
  ProdutoModel? produto;

  PecaModel({
    this.idPeca,
    this.numero,
    this.codigoFabrica,
    this.unidade,
    this.descricao,
    this.altura,
    this.largura,
    this.profundidade,
    this.unidadeMedida,
    this.volumes,
    this.active,
    this.custo,
    this.cor,
    this.material,
    this.idFornecedor,
    this.materialFabricacao,
    this.produto,
  });

   PecaModel.fromJson(Map<String, dynamic> json) {
    
      idPeca = json['idPeca'];
      numero = json['numero'];
      codigoFabrica = json['codigoFabrica'] ;
      unidade = json['unidade'];
      descricao = json['descricao'];
      altura = json['altura'];
      largura = json['largura'];
      profundidade =  json['profundidade'];
      unidadeMedida = json['unidadeMedida'];
      volumes =json['volumes'];
      active = json['active'];
      custo = json['custo'];
      cor = json['cor'];
      material =  json['material'];
      idFornecedor = json['idFornecedor'];
      materialFabricacao = json['materialFabricacao'];
      produto =  json['produto'] != null
          ? ProdutoModel.fromJson(json['produto'])
          : null;
    
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idPeca'] = this.idPeca;
    data['numero'] = this.numero;
    data['codigoFabrica'] = this.codigoFabrica;
    data['unidade'] = this.unidade;
    data['descricao'] = this.descricao;
    data['altura'] = this.altura;
    data['largura'] = this.largura;
    data['profundidade'] = this.profundidade;
    data['unidadeMedida'] = this.unidadeMedida;
    data['volumes'] = this.volumes;
    data['active'] = this.active;
    data['custo'] = this.custo;
    data['cor'] = this.cor;
    data['material'] = this.material;
    data['idFornecedor'] = this.idFornecedor;
    data['materialFabricacao'] = this.materialFabricacao;
    if (this.produto != null) {
      data['produto'] = this.produto?.toJson();
    }
    return data;
  }
}
